client.go

```diff
@@ -67,6 +67,10 @@ func HTTPClient(httpClient *http.Client) ClientParam {
 }
 
 func (c *Client) doRequest(method string, path string, params url.Values, body io.Reader) (*http.Response, error) {
+	return c.doRequestWithToken(method, path, params, body, c.authToken)
```

Comment:
Given you can delete a session token that isn't necessarily the token set during client creation, I thought it would be better to set `X-Sf-Token` at request time, rather than modify the clients underlying token before making the request, which could be problematic if calls are made in parallel.

Reply:
Yeah, makes sense! I didn't know this was how the session token API worked so it never occurred to me.